sycl/source/CMakeLists.txt

```diff
@@ -56,8 +63,9 @@ add_library(sycl SHARED
   "sampler.cpp"
   "stream.cpp"
   "spirv_ops.cpp"
+  "${OS_PLUGIN_INTERFACE}"
```

Comment:
`"${OS_PLUGIN_INTERFACE}"` could be replaced with a generator expression and the above `if-else` could be removed (untested):

```sh
$<$<PLATFORM_ID:Windows>:detail/windows_pi.cpp>
$<$<PLATFORM_ID:Linux>:detail/linux_pi.cpp>

```